### ISSUE TITLE
`pack create-builder` should add the lifecycle to the builder image

### ISSUE THREAD

**Issue by ekcasey** [buildpack/pack#110]
### Lifecycle Release Reference
**Given** I have a `builder.toml` with the following data
```
[lifecycle]
version = "v0.1.0"
```
**When** I run `pack create-builder my-builder -b builder.toml`
**Then** pack adds the lifecycle from the `v0.1.0` github release to `my-builder`
**When** I run `pack inspect-builder my-builder`
**Then** I observe `Lifecycle Version: v0.1.0` after `Stack:` in the output

Implementation Notes:
* lifecycle version should be added to the `io.buildpacks.builder.metadata` label
* downloaded `lifecycle.tgz` should be cached for preformance

### Lifecycle Path Reference w/ Version
**Given** a have a `builder.toml` with the following data
```
[lifecycle]
uri = "file://path/to/lifecycle.tgz"
version = "v0.2.0"
```

**When** I run `pack create-builder my-builder -b builder.toml`
**Then** pack adds the lifecycle from the `lifecycle.tgz` file to `my-builder` (paths are assumed to be relative to `builder.toml`)
**When** I run `pack inspect-builder my-builder`
**Then** I observe `Lifecycle Version: v0.2.0` after `Stack:` in the output
**When** I run `pack build --builder my-builder`

### Lifecycle Path Reference w/o Version
**Given** a have a `builder.toml` with the following data
```
[lifecycle]
uri = "file://path/to/lifecycle.tgz"
```

**When** I run `pack create-builder my-builder -b builder.toml`
**Then** pack adds the lifecycle from the `lifecycle.tgz` file to `my-builder` (paths are assumed to be relative to `builder.toml`)
**When** I run `pack inspect-builder my-builder`
**Then** I observe `Lifecycle Version: unknown` after `Stack:` in the output
**When** I run `pack build --builder my-builder`
**Then** I see a warning `running with unknown lifecycle` in my build output

### Default Lifecycle
**Given** I have a version of `pack` built to have default lifecycle version `v0.1.0`
**Given** I have a `builder.toml` without a `[lifecycle]` section
**When** I run `pack create-builder my-builder -b builder.toml`
**Then** pack adds the default lifecycle  to `my-builder`
**When** I run `pack inspect-builder my-builder`
**Then** I observe `Lifecycle Version: v0.1.0` after `Stack:` in the output

## Lifecycle
Remove all stack image building logic from `buildpack/lifecycle`
`Makefile` in lifecycle repo should build a `.tgz` of lifecycle binaries

## CI
Concourse should build stack images without baked in lifecycle binaries
Concourse should use a build flag to configure `pack` default lifecycle version to be the latest released lifecycle

**Comment by djoyahoy** [comment]
As an alternative to case 1 (using a `version` field`), I suggest that users can provide a `uri` to a github release and the version will be pulled from the released artifact. In this way, we could be consistent with how we select buildpacks in the `builder.toml`. This would require a `lifecycle.toml` or some kind of metadata in the released artifact to contain the version.

**Comment by jkutner** [comment]
How much (if at all) do we care about substituting some other spec impl for buildpack/lifecycle? Or should we just cross that bridge when we get there?

**Comment by ekcasey** [comment]
@jkutner given that pack depends on some unspec-ed features of lifecycle (the `cacher` and `restorer` for example), it seems unlikely to me that `pack` would be compatible with another lifecycles even if it adhered to the spec.

I'm not totally convinced that's a problem? I feel like it might be okay if a given platform has a contract with it's lifecycle that is a superset of the spec. Not sure what others think... 

We could further specify the platform contract. `analzyer` and `exporter` flags aren't included and I think maybe they should be. Things like the `-image` flag on `cacher` feel weirder though, that seems platform specific.

**Comment by squee1945** [comment]
The /lifecycle/launcher is also needed in the runtime image (another surprise to me), and it seems like this should be consistent with the lifecycle artifacts used in the builder. Any thought to automatically layering the launcher into the final built image following the builder.toml indication as well?

**Comment by ekcasey** [comment]
@squee1945 the `exporter` lifecycle phase already adds the `launcher` as a layer to the exported image. Some of the run images folks are using do have a `launcher` built in (for historical reasons) but this is unnecessary as of  https://github.com/buildpack/roadmap/issues/21 (https://github.com/buildpack/lifecycle/pull/59)  and the launcher will be overwritten. This means that rebasing won't break your app even if the interface between the launcher and the app changes.

**Comment by mgibson1121** [comment]
This story passes acceptance

**Comment by ekcasey** [comment]
implemented on `master` will ship in `v0.2.0`
